src/transformers/models/bart/modeling_bart.py

```diff
@@ -128,12 +128,14 @@ def __init__(self, num_embeddings: int, embedding_dim: int):
         self.offset = 2
         super().__init__(num_embeddings + self.offset, embedding_dim)
 
-    def forward(self, input_ids_shape: torch.Size, past_key_values_length: int = 0):
-        """`input_ids_shape` is expected to be [bsz x seqlen]."""
-        bsz, seq_len = input_ids_shape[:2]
+    def forward(self, input_ids: torch.Tensor, past_key_values_length: int = 0):
+        """`input_ids' shape is expected to be [bsz x seqlen]."""
+
+        bsz, seq_len = input_ids.shape[:2]
         positions = torch.arange(
             past_key_values_length, past_key_values_length + seq_len, dtype=torch.long, device=self.weight.device
-        )
+        ).repeat(bsz, 1)
```

Comment:
After looking into it, it comes from this `repeat`.
The error you get is something like "repeat: list index out of range", because for some reason, during tracing the `repeat` call becomes this:
```
kind = 'call_method'
target = 'repeat'
args = (Proxy(arange), Proxy(getitem_2), 1)
```
It is as if it was considering `self` (here `Proxy(arange)`) as a size in `torch.Tensor.repeat`...
Because this operation is failing, the tracer cannot compute the metadata needed to handle control-flow, hence your error.

In any case, I was able to fix it by doing this:
```suggestion
        ).expand(bsz, -1)
```
Which I think is better anyway because `expand` does not copy data, but maybe you want to copy stuff? I don't have the big picture here.

Reply:
Thank you so very much for looking into this @michaelbenayoun ! Great find. The use case does not care about copying stuff, so +1 for the more efficient method! :+1: